hive-event-listeners/apiary-gluesync-listener/src/main/java/com/expediagroup/apiary/extensions/gluesync/listener/ApiaryGlueSync.java

```diff
@@ -178,25 +179,88 @@ public void onAlterTable(AlterTableEvent event) {
     if (!event.getStatus()) {
       return;
     }
-    Table table = event.getNewTable();
+    Table oldTable = event.getOldTable();
+    Table newTable = event.getNewTable();
     try {
+      // Table rename are not supported by Glue, so we need to delete table and create again
+      if (isTableRename(oldTable, newTable)) {
+        log.info("{} glue table rename detected to {}", oldTable.getTableName(), newTable.getTableName());
+        createTable(newTable);
+        copyPartitions(newTable, getPartitions(oldTable));
```

Comment:
Wondering if should clean orphan resources in case the rename operation fails, but it might be difficult because glue client does not have checked exceptions.

Reply:
My current expectation is that in case glue is not able to make this rename, the whole hive operation will fail.

Reply:
This could be trouble if the Hive operation is fine but there's a bug/unexpected state in Glue Sync. 
I guess given we don't have retries it's the best we can do right now.